src/modules/navigator/rtl.cpp

```diff
@@ -68,6 +69,79 @@ void
 RTL::on_activation()
 {
 
+	// find the RTL destination: go through the safe points & home position
+	const home_position_s &home_position = *_navigator->get_home_position();
+	const vehicle_global_position_s &global_position = *_navigator->get_global_position();
+
+	mission_stats_entry_s stats;
+	int ret = dm_read(DM_KEY_SAFE_POINTS, 0, &stats, sizeof(mission_stats_entry_s));
+	int num_safe_points = 0;
+
+	if (ret == sizeof(mission_stats_entry_s)) {
+		num_safe_points = stats.num_items;
+	}
+
+	int closest_index = 0;
+	mission_safe_point_s closest_safe_point;
+
+	// take home position into account
+	double dlat = home_position.lat - global_position.lat;
+	double dlon = home_position.lon - global_position.lon;
+	double min_dist_squared = dlat * dlat + dlon * dlon;
+
+	// find the closest point
+	for (int current_seq = 1; current_seq <= num_safe_points; ++current_seq) {
+		mission_safe_point_s mission_safe_point;
+
+		if (dm_read(DM_KEY_SAFE_POINTS, current_seq, &mission_safe_point, sizeof(mission_safe_point_s)) !=
+		    sizeof(mission_safe_point_s)) {
+			PX4_ERR("dm_read failed");
+			continue;
+		}
+
+		// TODO: take altitude into account for distance measurement
+
+		dlat = mission_safe_point.lat - global_position.lat;
+		dlon = mission_safe_point.lon - global_position.lon;
+		double dist_squared = dlat * dlat + dlon * dlon;
+
+		if (dist_squared < min_dist_squared) {
+			closest_index = current_seq;
+			min_dist_squared = dist_squared;
+			closest_safe_point = mission_safe_point;
+		}
+	}
+
+	if (closest_index == 0) {
+		_destination.set(home_position);
+
+	} else {
+
+		// TODO: handle all possible mission_safe_point.frame cases
+		switch (closest_safe_point.frame) {
+		case 0:
```

Comment:
I think it's worth commenting which on it is:
```suggestion
		case 0: // MAV_FRAME_GLOBAL
```
Or turning `mission_safe_point_s::frame` into an enum.